bulldozer/config_v1.go

```diff
@@ -34,8 +34,10 @@ const (
 )
 
 type MergeConfig struct {
-	Whitelist Signals `yaml:"whitelist"`
-	Blacklist Signals `yaml:"blacklist"`
+	Allowlist Signals  `yaml:"allowlist"`
+	Denylist  Signals  `yaml:"denylist"`
+	Blacklist *Signals `yaml:"blacklist,omitempty"`
+	Whitelist *Signals `yaml:"whitelist,omitempty"`
```

Comment:
I would prefer something like.. 
```
	Allowlist Signals  `yaml:"allowlist,whitelist"`
```

Reply:
An alternate option might be to only define the new fields and them implement a custom `UnmarshalYAML` function for these types. That would probably do something like:

1. Unmarshal into the full struct with the new fields
2. Try to unmarshal into a subset struct containing only old fields:
   ```go
   var legacy struct {
     Blacklist *Signals `yaml:"blacklist"`
     Whitelist *Signals `yaml:"whitelist"`
   }
   if err := unmarshal(&legacy); err != nil {
     // ...
   }
3. Merge the fields into the main struct if they are set

This all requires that the `unmarshal` function passed to `UnmarshalYAML` is non-strict (i.e. allows unknown fields), but I think that is the case.